### PR TITLE
Rewrite JSON.stringify description

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

There was a lot of description buried in the examples section, which isn't very discoverable (especially after the giant code blocks).

#### Metadata

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/json/stringify/index.md [hunk 1]
```diff
@@ -33,230 +31,262 @@ JSON.stringify(value, replacer, space)
...
+/*
+{
+ "a": 2
+}
+*/
 ```
 
 Using a tab character mimics standard pretty-print appearance:
 
+<!-- markdownlint-disable MD010 -->
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:271]: This is a bit annoying but understandable.

File: files/en-us/web/javascript/reference/global_objects/json/stringify/index.md [hunk 2]
```diff
@@ -33,230 +31,263 @@ JSON.stringify(value, replacer, space)
...
 - `space` {{optional_inline}}
 
-  - : A {{JSxRef("String")}} or {{JSxRef("Number")}} object that's used to insert white space (including indentation, line break characters, etc.) into the output JSON string for readability purposes.
+  - : A string or number that's used to insert white space (including indentation, line break characters, etc.) into the output JSON string for readability purposes.
 
-    If this is a `Number`, it indicates the number of space characters to
-    use as white space for indenting purposes; this number is capped at 10 (if it is greater, the value is just
-    `10`). Values less than 1 indicate that no space should be used.
+    If this is a number, it indicates the number of space characters to be used as indentation; this number is capped at 10 (if it is greater, the value is just `10`). Values less than 1 indicate that no space should be used.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:39]:
```suggestion
    If this is a number, it indicates the number of space characters to be used as indentation, clamped to 10 (that is, any number greater than `10` is treated as if it were `10`). Values less than 1 indicate that no space should be used.
```

File: files/en-us/web/javascript/reference/global_objects/json/stringify/index.md [hunk 3]
```diff
@@ -33,230 +31,263 @@ JSON.stringify(value, replacer, space)
...
-- All the other {{JSxRef("Object")}} instances (including {{JSxRef("Map")}},
-  {{JSxRef("Set")}}, {{JSxRef("WeakMap")}}, and {{JSxRef("WeakSet")}}) will have only
-  their enumerable properties serialized.
-  - Enumerable properties are visited using the same algorithm as [`Object.keys`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys), which has a well-defined order and is stable across implementations. For example, `JSON.stringify` on the same object will always produce the same string, and `JSON.parse(JSON.stringify(obj))` would produce an object with the same key ordering as the original (assuming the object is completely JSON-serializable).
+- {{JSxRef("Boolean")}}, {{JSxRef("Number")}}, {{JSxRef("String")}}, and {{jsxref("BigInt")}} (obtainable via [`Object()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/Object)) objects are converted to the corresponding primitive values during stringification, in accord with the traditional conversion semantics. {{jsxref("Symbol")}} objects (obtainable via [`Object()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/Object)) are treated as plain objects.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:60]:
```suggestion
- {{JSxRef("Boolean")}}, {{JSxRef("Number")}}, {{JSxRef("String")}}, and {{jsxref("BigInt")}} (obtainable via [`Object()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/Object)) objects are converted to the corresponding primitive values during stringification, in accordance with the traditional conversion semantics. {{jsxref("Symbol")}} objects (obtainable via [`Object()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/Object)) are treated as plain objects.
```

File: files/en-us/web/javascript/reference/global_objects/json/stringify/index.md [hunk 4]
```diff
@@ -33,230 +31,263 @@ JSON.stringify(value, replacer, space)
...
+Typically, array elements' index would never shift (even when the element is an invalid value like a function, it will become `null` instead of omitted). Using the `replacer` function allows you to control the order of the array elements by returning a different array.
+
+### The space argument
+
+The `space` argument may be used to control spacing in the final string.
+
+- If it is a number, successive levels in the stringification will each be indented by this many space characters.
+- If it is a string, successive levels will be indented by this string.
+
+Each level of indentation will never be longer than 10. Number values of `space` are capped at 10, and string values are truncated to 10 characters.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:106]:
```suggestion
Each level of indentation will never be longer than 10. Number values of `space` are clamped to 10, and string values are truncated to 10 characters.
```

File: files/en-us/web/javascript/reference/global_objects/json/stringify/index.md [hunk 5]
```diff
@@ -288,88 +316,34 @@ circularReference.myself = circularReference;
...
-```
+If you are using `JSON.stringify()` to deep-copy an object, you may want to use [`structuredClone()`](/en-US/docs/Web/API/structuredClone) instead, which supports circular references. JavaScript engine APIs for binary serialization, such as [`v8.serialize()`](https://nodejs.org/api/v8.html#v8serializevalue), also support circular references.
 
-### Example of using JSON.stringify() with localStorage
+### Using JSON.stringify() with localStorage
 
-In a case where you want to store an object created by your user and allowing it to be
-restored even after the browser has been closed, the following example is a model for
-the applicability of `JSON.stringify()`:
+In a case where you want to store an object created by your user and allowing it to be restored even after the browser has been closed, the following example is a model for the applicability of `JSON.stringify()`:
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:325]:
```suggestion
In a case where you want to store an object created by your user and allow it to be restored even after the browser has been closed, the following example is a model for the applicability of `JSON.stringify()`:
```

File: files/en-us/web/javascript/reference/global_objects/json/stringify/index.md [hunk 6]
```diff
@@ -288,88 +316,34 @@ circularReference.myself = circularReference;
...
-const a = JSON.stringify({ foo: "bar", baz: "quux" })
-//'{"foo":"bar","baz":"quux"}'
-const b = JSON.stringify({ baz: "quux", foo: "bar" })
-//'{"baz":"quux","foo":"bar"}'
-console.log(a !== b) // true
-
-// some memoization functions use JSON.stringify to serialize arguments,
-// which may cause a cache miss when encountering the same object like above
-```
+If you are using `JSON.stringify()` to deep-copy an object, you may want to use [`structuredClone()`](/en-US/docs/Web/API/structuredClone) instead, which supports circular references. JavaScript engine APIs for binary serialization, such as [`v8.serialize()`](https://nodejs.org/api/v8.html#v8serializevalue), also support circular references.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:321]:
```suggestion
If you are using `JSON.stringify()` to deep-copy an object, you may instead want to use [`structuredClone()`](/en-US/docs/Web/API/structuredClone), which supports circular references. JavaScript engine APIs for binary serialization, such as [`v8.serialize()`](https://nodejs.org/api/v8.html#v8serializevalue), also support circular references.
```

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**